source/integrate/jira-interoperability.rst

```diff
@@ -0,0 +1,298 @@
+
+Install Jira integration in your Jira instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable your users to create and manage Jira issues across Mattermost channels, you must install this Jira integration, as an application, in your Jira instance. 
+
+- For Jira Server or Data Center instances, run the ``/jira instance install server <YOUR-JIRA-URL>`` slash command in a Mattermost channel as a Mattermost system admin, then follow the steps posted to the channel, replacing ``YOUR-JIRA-URL`` with your Jira URL. This value must match the Jira server URL you use to log in. Run the ``/jira instance uninstall server <YOUR-JIRA-URL>`` slash command to disconnect Mattermost from your Jira Server or Data Center instance.
+- For Jira Cloud, run the  ``/jira instance install cloud <YOUR-JIRA-URL>`` slash command. Run the ``/jira instance uninstall cloud <YOUR-JIRA-URL>`` slash command to disconnect Mattermost from your Jira Cloud instance.
+- You must explicitly share the integration in your Jira instance by going to https://developer.atlassian.com/console/myapps, selecting **Distribution**, and sharing it with the organization.
+
+Configure webhooks in Jira
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A Mattermost system admin and a Jira system admin must perform the following steps to configure a single webhook for all possible event triggers, called a firehose, that you would like to be pushed into Mattermost. Mattermost receives a stream of events from the Jira server via a configured webhook, and routes the events to specific channels. A `channel subscription <#manage-channel-subscriptions-in-mattermost>`__ processes the firehose of data, and routes the events to channels based on your subscriptions. 
+
+1. In Mattermost, run the  ``/jira webhook <YOUR-JIRA-URL>`` slash command in a Mattermost channel to get the appropriate webhook URL, replacing ``YOUR-JIRA-URL`` with your Jira URL.
+2. In Jira, go to **Jira Settings > System > WebHooks**. (For older versions of Jira, select the gear icon in bottom left corner, then go to **Advanced > WebHooks**.)
+3. Select **Create a WebHook**.
+4. Enter a **Name** for the webhook and add the Jira webhook URL you retrieved above in Mattermost as the URL.
+5. Specify the issue events that will be sent to Mattermost channels by selecting all of the following: 
+
+  - Comments: created, updated, and deleted.
+  - Issues: created, updated, and deleted.
+
+6. Select **Save**.
+
+Legacy Jira webhooks
+^^^^^^^^^^^^^^^^^^^^
+
+If your organization's infrastructure is set up in such a way that your Mattermost instance can't connect to your Jira instance, you won't be able to subscribe Mattermost channels. You'll need to use legacy webhooks instead.
+
+1. To generate the webhook URL for a specific channel, run the  ``/jira webhook`` slash command, and use the URL output in the **Legacy Webhooks** section of the output.
+
```

Comment:
I think this section could be de-emphasized, and put after the regular subscriptions section. Unrelated to the purpose of the PR though

Reply:
A very good piece of feedback that I'll action. Thanks!